File: 2021/python/day1.py
with open('../input/day1.txt') as file:
    myList = [int(line.strip()) for line in file]

last1 = None
last2 = None
last3 = None
lastSum = None

partOne = 0
partTwo = 0

for item in myList:
    if last1 is None:
        last1 = item
    elif last2 is None:
        last2 = item
    elif last3 is None:
        last3 = item
        lastSum = last1 + last2 + last3
    else:
        last1 = last2
        last2 = last3
        last3 = item
        newSum = last1 + last2 + last3
        if newSum > lastSum:
            partTwo += 1
        lastSum = newSum
    
    if last2 is not None:
        if last2 > last1:
            partOne += 1

print("partOne = " + str(partOne))
print("partTwo = " + str(partTwo))

# $ python3 day1.py
# partOne = 1316
# partTwo = 1344

File: 2021/rust/day1.rs
use std::fs::File;
use std::io::BufReader;
use std::io::prelude::*;

fn main() {
    let f = File::open("../input/day1.txt").expect("Error opening file");
    let my_list = BufReader::new(f);

    let mut last1: i64 = -1;
    let mut last2: i64 = -1;
    let mut last3: i64 = -1;

    let mut last_sum: i64 = -1;
    let mut new_sum: i64;

    let mut part_one: i64 = 0;
    let mut part_two: i64 = 0;
    
    for item in my_list.lines() {
        if last1 == -1 {
            last1 = item.unwrap().parse::<i64>().unwrap();
        } else if last2 == -1 {
            last2 = item.unwrap().parse::<i64>().unwrap();
        } else if last3 == -1 {
            last3 = item.unwrap().parse::<i64>().unwrap();
        } else {
            last1 = last2;
            last2 = last3;
            last3 = item.unwrap().parse::<i64>().unwrap();
            new_sum = last1 + last2 + last3;
            if new_sum > last_sum {
                part_two += 1;
            }
            last_sum = new_sum;
        }

        if last2 != -1 {
            if last2 > last1 {
                part_one += 1;
            }
        }
    }

    println!("part_one = {}", part_one);
    println!("part_two = {}", part_two);
}

// $ rustc day1.rs
// $ ./day1
// part_one = 1316
// part_two = 1344